framework/deproxy_client.py

```diff
@@ -1,72 +1,242 @@
 
     def recieve_response(self, response):
         tf_cfg.dbg(4, "Recieved response: %s" % str(response.msg))
         self.responses.append(response)
-        self.last_response = response
-
-    def make_request(self, request):
-        self.nr = len(self.responses)
-        BaseDeproxyClient.make_request(self, request)
 
     def wait_for_response(self, timeout=5):
-        if self.state != stateful.STATE_STARTED:
-            return False
-
         t0 = time.time()
-        while len(self.responses) == self.nr:
+        while len(self.responses) == self.__nr:
             t = time.time()
             if t - t0 > timeout:
                 return False
             time.sleep(0.01)
         return True
+
+    def make_connect(self, addr, port):
+        self.connect((addr, port))
+
+    # stateful methods
+    def __stop_client(self):
+        tf_cfg.dbg(4, '\tStop deproxy client')
+        self.polling_lock.acquire()
+        tf_cfg.dbg(3, "\tDeproxy client stop: Locked acquired")
+
+        try:
+            self.close()
+        except Exception as e:
+            tf_cfg.dbg(2, "Error while stopping client: %s" % str(e))
+            self.polling_lock.release()
+            raise e
+
+        self.polling_lock.release()
+        tf_cfg.dbg(3, "\tDeproxy client stop: Locked released")
+
+    def run_start(self):
+        tf_cfg.dbg(3, '\tStarting deproxy client')
+        tf_cfg.dbg(4, '\tDeproxy: Client: Connect to %s:%d.' % \
+                   (self.addr, self.port))
+
+        self.polling_lock.acquire()
+
+        tf_cfg.dbg(3, "\tDeproxy client start: Locked acquired")
+        try:
+            self.init_socket()
+        except Exception as e:
+            tf_cfg.dbg(2, "Error while starting client: %s" % str(e))
+            self.polling_lock.release()
+            raise e
+        self.polling_lock.release()
+        tf_cfg.dbg(3, "\tDeproxy client start: Locked released")
+        self.make_connect(self.addr, self.port)
+    # end stateful methods
+
+class DeproxyClient(BaseAsyncoreClient):
+
+    last_response = None
+
+    def init_socket(self):
+        self.create_socket(socket.AF_INET, socket.SOCK_STREAM)
+
+    def recieve_response(self, response):
+        BaseAsyncoreClient.recieve_response(self, response)
+        self.last_response = response
+
+def create_client_deproxy(client):
+    addr = fill_template(client['addr'])
+    port = int(fill_template(client['port']))
+    clt = DeproxyClient(addr=addr, port=port)
+    return clt
+
+
+class SSLDeproxyClient(BaseAsyncoreClient):
+
+    def __init__(self, ca=None, *args, **kwargs):
+        self.ca_certs = ca
+        self.s = None
+        self.ssl_failed = False
+        BaseAsyncoreClient.__init__(self, *args, **kwargs)
+
+    def recieve_response(self, response):
+        BaseAsyncoreClient.recieve_response(self, response)
+
+    def init_socket(self):
+        self.s = socket.socket()
+        #self.ssl_socket = ssl.SSLSocket(sock=self.s,
```

Comment:
Seems like the code was commented out for the troubleshooting, but the right version wasn't returned back. Need to fix it.

Reply:
fixed, commented code is removed